### PR TITLE
simple benchmark tests

### PR DESCRIPTION
**Pull request by qingsenzhang**

### CODE REVIEW

File: benchmark_tests/qingsen-run.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+LAYERS=18
+WIDE_SCALE=8
+export MXNET_SWAP_NO_COPY=1
+export MXNET_SWAP_THRESHOLD_MULTIPLIER=512
+
+python benchmark.py --num_layers=${LAYERS} --batch_size 4 --wide_scale=${WIDE_SCALE} --num_loop=6 --cold_skip=3 resnet > log_resnet_${LAYERS}_4_${WIDE_SCALE}_no_copy
```
Comment (fegin) [inline benchmark_tests/qingsen-run.sh]: I don't think include the username to the filename is a good naming style unless you have a very good reason. run.sh may be enough.

File: benchmark_tests/benchmark.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                        args_grad=args_grad,
+                        grad_req='write',
+                        group2ctx=group2ctx)
+
+    feed_args(net, arg_arrays)
+    all_time = []
+    for i in range(num_loops):
+        print('=> loop %d' % i);
+	#uncomment this line to enable start_iteration()
+        #mx.base.start_iteration()
```
Comment (fegin) [inline benchmark_tests/benchmark.py]: The indentions are not consistent. This is not allowed by Python.

File: benchmark_tests/benchmark.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+from __future__ import print_function
+
+import mxnet as mx
+import numpy as np
+import os, sys,time
+import pickle as pickle
+import logging
+import argparse
+
+rng = np.random.RandomState(seed=42)
```
Comment (fegin) [inline benchmark_tests/benchmark.py]: This is not used. Please removed.

File: benchmark_tests/benchmark.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
+# pylint: skip-file
+from __future__ import print_function
+
+import mxnet as mx
+import numpy as np
+import os, sys,time
+import pickle as pickle
```
Comment (fegin) [inline benchmark_tests/benchmark.py]: This is not used. Please removed.

File: benchmark_tests/benchmark.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                 if name != 'data' and not name.endswith('label')}
+    '''
+    size = 0
+    for name, shape in zip(net.list_arguments(), arg_shapes):
+        if 'weight' in name:
+            size += np.prod(shape) * 4
+        print (name, np.prod(shape))
+    print('size {}'.format(size / 1024.0 / 1024.0 / 1024.0))
+    assert False
+    '''
```
Comment (fegin) [inline benchmark_tests/benchmark.py]: Removed this comment too.

File: benchmark_tests/qingsen-run.sh [hunk 6]
```diff
@@ -0,0 +1,6 @@
+LAYERS=18
+WIDE_SCALE=8
+export MXNET_SWAP_NO_COPY=1
+export MXNET_SWAP_THRESHOLD_MULTIPLIER=512
```
Comment (fegin) [inline benchmark_tests/qingsen-run.sh]: These two do not exist in the current code base. You should remove them.

**Review by fegin — CHANGES_REQUESTED** [review]
@PhantomPhillipe Please review all the code again and remove unnecessary code and variables. I commented a few. When cleaning the code like we are doing now, you should always start with the minimum set. If we need those functions/variables in the future, we can add them back. Too many redundant codes make it hard to maintain.

File: benchmark_tests/benchmark.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+            t0 = time.time()
+        outputs = executor.forward()
+        if num_classes is None:
+          executor.backward(outputs[0])
+        else:
+          executor.backward()
+        for name, grad in args_grad.items():
+            grad.wait_to_read()
+        #uncomment this line to enable stop_iteration()
+	#mx.base.stop_iteration()
```
Comment (fegin) [inline benchmark_tests/benchmark.py:80]: The indentions are not consistent. This is not allowed by Python.

**Comment by fegin** [comment]
Please also add @Sotskin and @KarlLeell as reviewers. I have no further comments. Please fix the issues I mentioned.

**Comment by qingsenzhang** [comment]
@fegin Modified accordingly. Please have a look.

**Comment by Sotskin** [comment]
Looks good now

**Review by Sotskin — APPROVED** [review]

**Review by KarlLeell — APPROVED** [review]
I have no comment to make.

**Comment by Sotskin** [comment]
I just realized the branch to merge into is not right. Should be gpu_swap instead of master

**Pull request merged**